### PR TITLE
Create GH Actions CI Pipeline w/ Expected Failed Tests

### PR DESCRIPTION
**Pull request by bcmeireles**
### Summary

Created a workflow to run the tests for `dbt-dremio` when a PR is created

### Description

Since we have 22 failing tests that we plan to work on fixing, the current pipeline will just ignore pytest's exit code on error and will compare the failing tests with the tests that are expected to fail.

The pipeline from https://github.com/dremio/dbt-dremio/pull/249 is to be deployed once all the tests are fixed and it does not use this comparison approach

The test suites are also ran separately (folder by folder) as some tests will fail when trying to run all tests consecutively from the root test folder

A list of expected failures can be found (and edited) in `.github/expected_failures.txt`

### Test Results

Failing (expected) tests:
```
tests/component/test_profile_template.py::TestProfileTemplate::test_cloud_options
tests/functional/adapter/dbt_clone/test_dbt_clone.py::TestCloneNotPossibleDremio::test_can_clone_false
tests/functional/adapter/dremio_specific/test_drop_temp_table.py::TestDropTempTableDremio::test_drop_temp_table
tests/functional/adapter/dremio_specific/test_schema_parsing.py::TestSchemaParsingDremio::test_schema_with_dots
tests/functional/adapter/dremio_specific/test_verify_ssl.py::TestVerifyCertificateDremio::test_insecure_request_warning_not_exist
tests/functional/adapter/grants/test_incremental_grants.py::TestIncrementalGrantsDremio::test_incremental_grants
tests/functional/adapter/grants/test_invalid_grants.py::TestInvalidGrantsDremio::test_invalid_grants
tests/functional/adapter/grants/test_model_grants.py::TestTableGrantsDremio::test_view_table_grants
tests/functional/adapter/grants/test_model_grants.py::TestViewGrantsDremio::test_view_table_grants
tests/functional/adapter/grants/test_seed_grants.py::TestSeedGrantsDremio::test_seed_grants
tests/functional/adapter/grants/test_snapshot_grants.py::TestSnapshotGrantsDremio::test_snapshot_grants
tests/functional/adapter/relation/test_get_relation_last_modified.py::TestGetLastRelationModified::test_get_last_relation_modified
tests/functional/adapter/unit_testing/test_unit_testing.py::TestDremioUnitTestCaseInsensitivity::test_case_insensitivity
tests/functional/adapter/unit_testing/test_unit_testing.py::TestDremioUnitTestInvalidInput::test_invalid_input
tests/functional/adapter/unit_testing/test_unit_testing.py::TestDremioUnitTestingTypes::test_unit_test_data_type
tests/hooks/test_model_hooks.py::TestPrePostModelHooksInConfigDremio::test_pre_and_post_model_hooks_model
tests/hooks/test_model_hooks.py::TestPrePostModelHooksInConfigKwargsDremio::test_pre_and_post_model_hooks_model
tests/hooks/test_model_hooks.py::TestPrePostModelHooksInConfigWithCountDremio::test_pre_and_post_model_hooks_model_and_project
tests/hooks/test_model_hooks.py::TestPrePostModelHooksOnSnapshotsDremio::test_hooks_on_snapshots
tests/hooks/test_model_hooks.py::TestPrePostSnapshotHooksInConfigKwargsDremio::test_hooks_on_snapshots
tests/hooks/test_run_hooks.py::TestPrePostRunHooksDremio::test_pre_and_post_run_hooks
tests/hooks/test_run_hooks.py::TestPrePostRunHooksDremio::test_pre_and_post_seed_hooks
```

### Changelog

- created `.github/workflows/ci.yml`

### Related Issue

https://github.com/dremio/dbt-dremio/issues/39

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+              echo "MinIO is up."
+              break
+            fi
+            echo "Waiting for MinIO to become ready..."
+            sleep 5
+          done
+          if ! docker exec minio curl -s http://localhost:9000/minio/health/live; then
+            echo "MinIO did not become ready in time."
+            exit 1
+          fi
```
Comment (simonpannek) [inline .github/workflows/ci.yml:75]: [SUGGESTION]
Can we store these retries and sleep values in some variables (maybe we can use ENV)

**Review by simonpannek — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+      - name: Install MinIO Client (mc)
+        run: |
+          curl -O https://dl.min.io/client/mc/release/linux-amd64/mc
+          chmod +x mc
+          sudo mv mc /usr/local/bin/
+
+      - name: Create MinIO bucket
+        run: |
+          for i in {1..12}; do
```
Comment (howareyouman) [inline .github/workflows/ci.yml:64]: [NIT] Do we need this hacky `1..12` and `sleep 5` for our CI?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+          echo "Listing all buckets to verify"
+          mc ls myminio
+
+      - name: "Create Dremio S3 Source"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y curl jq
+
+          for i in {1..12}; do
```
Comment (howareyouman) [inline .github/workflows/ci.yml:94]: [SUGGESTION] Copy/paste function for some services to be alive. Should we create a function and reuse it?

**Review by howareyouman — APPROVED** [review]
The automation code is sometimes not that readable, but at least it works and that is great!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (bcmeireles) [reply .github/workflows/ci.yml:64]: I've changed the approach to how @simonpannek suggested, I'm waiting 5 secs between pings to not spam the service and giving it a total of 1 minute to ensure it started successfully before failing the job

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bcmeireles) [reply .github/workflows/ci.yml:94]: Only the loop itself is repeated only once, with the env vars change it should be cleaner

**Comment by bcmeireles** [comment]
> The automation code is sometimes not that readable, but at least it works and that is great!

Hopefully once we fix the remaining tests and get to finish and merge https://github.com/dremio/dbt-dremio/pull/249 we'll have get rid of most of the "hacks"

**Pull request merged**